### ISSUE TITLE
Batch_size in train.py should adjust dynamically, instead of being fixed.

### ISSUE THREAD

**Issue by Bjohnson131** [qqwweee/keras-yolo3#561]
To calculate batch size, one can use a script like so.  by multiplying the input shape's area by the number of channels, and then by a factor of 1050 (or about 800 for Tiny yolo), you can optimize the batch size for various training setups.

```
System_memory = 16,000,000,000 #System memory in bytes. In this case, 16Gb
input_shape = (416,416)

batch_size = min((input_shape[0]*input_shape[1]*3*1050)/System_memory,32)
```

**Comment by kidboy-man** [comment]
@tuskiomi what is the effect of reducing/increasing batch size? and which one we should attempt to achieve higher accuracy? Many thanks

**Comment by Bjohnson131** [comment]
> @tuskiomi what is the effect of reducing/increasing batch size? and which one we should attempt to achieve higher accuracy? Many thanks

Generally the batch size should be as large as your memory allows for improved generalization
